rpc_jobs/build_gating_venv.yml

```diff
@@ -20,102 +20,99 @@
         lock('build_gating_venv'){
           node('pubcloud_multiuse'){
             try{
-              deleteDir()
-              common.clone_with_pr_refs('rpc-gating',
-                                        'git@github.com:rcbops/rpc-gating',
-                                        env.RPC_GATING_BRANCH)
-              dir('rpc-gating'){
-                // run in docker to ensure all the required apt packages are available
-                container = docker.build env.BUILD_TAG.toLowerCase()
-              }
-              // Venv is build within docker as it requires extra packages
-              container.inside{
-                sh """#!/bin/bash -xeu
+              withCredentials(artifact_build.get_rpc_repo_creds()) {
+
+                deleteDir()
+                common.clone_with_pr_refs('rpc-gating',
+                                          'git@github.com:rcbops/rpc-gating',
+                                          env.RPC_GATING_BRANCH)
+                dir('rpc-gating'){
+                  // run in docker to ensure all the required apt packages are available
+                  container = docker.build(env.BUILD_TAG.toLowerCase())
+                }
 
-                # Install virtualenv if required
-                if [[ ! -d ".venv" ]]; then
-                  requirements="virtualenv==15.1.0"
-                  pip install -U "\${requirements}" \
-                    || pip install --isolated -U "\${requirements}"
+                // Venv is built within docker, includes all the environment variables
+                container.inside{
+                  sshagent(credentials:['rpc_repo_ssh_user_private_key']){
```

Comment:
Friday afternoon brain here, but I don't see this credential.  Should this be `RPC_REPO_SSH_USER_PRIVATE_KEY_FILE`?

Reply:
Similar to @git-harry's comment above, I'd probably only wrap the ssh commands in the sshagent block to reduce the time the agent's running.